pkg/api/componentreadiness/component_report.go

```diff
@@ -1321,6 +1579,96 @@ func (c *componentReportGenerator) getRequiredConfidence(testID string, variants
 	return c.RequestAdvancedOptions.Confidence
 }
 
+func (c *componentReportGenerator) matchBaseRegression(testID crtype.ReportTestIdentification, baseRelease string, baseStats crtype.TestStatus) (crtype.TestStatus, string) {
```

Comment:
Taking a pass at godoc would be helpful on some of these more important functions.